op-chain-ops/cmd/celo-migrate/ancients.go

```diff
@@ -23,6 +24,44 @@ type RLPBlockRange struct {
 	tds      [][]byte
 }
 
+// CheckRLPBlockRangeForGaps checks for gaps in the given RLP block range by comparing the lengths for each table and checking the header numbers
+func CheckRLPBlockRangeForGaps(blockRange RLPBlockRange, expectedLength uint64) (err error) {
+	if uint64(len(blockRange.hashes)) != expectedLength {
+		err = fmt.Errorf("Expected count mismatch in block range hashes: expected %d, actual %d", expectedLength, len(blockRange.hashes))
+	}
+	if uint64(len(blockRange.bodies)) != expectedLength {
+		err = errors.Join(err, fmt.Errorf("Expected count mismatch in block range bodies: expected %d, actual %d", expectedLength, len(blockRange.bodies)))
+	}
+	if uint64(len(blockRange.headers)) != expectedLength {
+		err = errors.Join(err, fmt.Errorf("Expected count mismatch in block range headers: expected %d, actual %d", expectedLength, len(blockRange.headers)))
+	}
+	if uint64(len(blockRange.receipts)) != expectedLength {
+		err = errors.Join(err, fmt.Errorf("Expected count mismatch in block range receipts: expected %d, actual %d", expectedLength, len(blockRange.receipts)))
+	}
+	if uint64(len(blockRange.tds)) != expectedLength {
+		err = errors.Join(err, fmt.Errorf("Expected count mismatch in block range total difficulties: expected %d, actual %d", expectedLength, len(blockRange.tds)))
+	}
+
+	if err != nil {
+		return err
+	}
+
+	// Cbecm that block number in header matches the expected block number
+	for i := uint64(0); i < expectedLength; i++ {
+		header := new(types.Header)
+		err := rlp.DecodeBytes(blockRange.headers[i], &header)
+		if err != nil {
+			return fmt.Errorf("can't decode header: %w", err)
+		}
+		expectedBlockNumber := blockRange.start + i
+		if header.Number.Uint64() != expectedBlockNumber {
+			return fmt.Errorf("header number mismatch: expected %d, actual %d", expectedBlockNumber, header.Number.Uint64())
+		}
```

Comment:
I would also check for parent header consistency, so check that the parentHash of block `N` equals the hash of  block `N-1`.

Reply:
So I added this check and it appears the parent hashes don't match the previous block hash. I guess this makes sense, as we've changed the format of the header but not the ParentHash field. @piersy Do you think this is something we should transform in the block header or leave as is?

Reply:
Nevermind, figured out that this mismatch was due to hashing over the non-transformed header. I fixed the check and added a comment explaining this.